temoa/extensions/framework.py

```
@@ -0,0 +1,237 @@
+def normalize_extension_ids(extension_ids: Sequence[str | object] | None) -> tuple[str, ...]:
+    """Normalize configured extension ids while preserving user-provided order."""
+    if not extension_ids:
+        return ()
+
+    normalized: list[str] = []
+    seen: set[str] = set()
+    for ext_id in extension_ids:
+        if not isinstance(ext_id, str):
+            msg = f'Extension ids must be strings. Received: {type(ext_id).__name__}'
+            logger.error(msg)
+            raise ValueError(msg)
```

Ruff (0.15.21)

[warning] Prefer `TypeError` exception for invalid type

(TRY004)